src/pz390.java

```diff
@@ -9233,6 +9269,22 @@ private void ins_E3XX(){   // RPI 1207
 			ins_setup_rxy();
 			reg.putInt(rf1 + 4, reg.getInt(rf1 + 4) * mem.getInt(xbd2_loc));
 			break;
+		case 0x53: // "E353" "MSC" "RXY" // #330
+		    psw_check = false;
+		    ins_setup_rxy();
+			rlv1 = (long)reg.getInt(rf1 + 4) * (long)mem.getInt(xbd2_loc);
+			rv1  = (int)rlv1;
+			reg.putInt(rf1 + 4, rv1);
+			if (rlv1 != (long)rv1){
+			  psw_cc = psw_cc3;
+			} else if (rv1 > 0){
+			  psw_cc = psw_cc2;
+			} else if (rv1 == 0) {
+			  psw_cc = psw_cc0;
+			} else {
+              psw_cc = psw_cc1;
+			}
+		    break; 
```

Comment:
See MSGRKC comment about setting the condition code and possible fixed-point overflow.